docs/topics/db/managers.txt

```diff
@@ -66,36 +66,21 @@ returns a list of all ``OpinionPoll`` objects, each with an extra
 
     class PollManager(models.Manager):
         def with_counts(self):
-            from django.db import connection
-            with connection.cursor() as cursor:
-                cursor.execute("""
-                    SELECT p.id, p.question, p.poll_date, COUNT(*)
-                    FROM polls_opinionpoll p, polls_response r
-                    WHERE p.id = r.poll_id
-                    GROUP BY p.id, p.question, p.poll_date
-                    ORDER BY p.poll_date DESC""")
-                result_list = []
-                for row in cursor.fetchall():
-                    p = self.model(id=row[0], question=row[1], poll_date=row[2])
-                    p.num_responses = row[3]
-                    result_list.append(p)
-            return result_list
+            return self.annotate(num_responses=Count("response"))
```

Comment:
I'm not sure if a note before this example is still correct:
```
A custom ``Manager`` method can return anything you want. It doesn't have to
return a ``QuerySet``.

For example, this custom ``Manager`` offers a method ``with_counts()``, which
returns a list of all ``OpinionPoll`` objects, each with an extra
``num_responses`` attribute that is the result of an aggregate query::
```
The new example returns a queryset.

Reply:
👀 👍

Reply:
Fixed. I also put a `Coalesce` on the `Count()` so it would return `0` instead of `None` because types.

Reply:
The note above the example is still outdated.

Reply:
Oh yes - and it was a bit redundant. Tidied up by keeping the explanation only at the end.